remove bundled pyh5md

XzzX opened the pull request:
pyh5md can be installed via pip, no need to bundle it

junghans reviewed (changes requested):
CMake needs to check if the `pyh5md` module is actually there.

junghans commented:
Aren't the tests using pyh5md?

XzzX commented:
> Aren't the tests using pyh5md?

Yes, the python tests do.

XzzX commented:
I reworked the readme and introduced a requirements.txt file. I think something like `could not find xxx` while executing the tests is as good as checking it via CMake.

junghans commented:
yeah, as long as the test don't fail, but get skipped everything is fine.

XzzX commented:
> yeah, as long as the test don't fail, but get skipped everything is fine.

They will fail. But I think this is ok since you do not meet the dependencies. C++ will also not build if you do not have the dependencies installed.

junghans commented:
> > yeah, as long as the test don't fail, but get skipped everything is fine.
> 
> They will fail. But I think this is ok since you do not meet the dependencies. C++ will also not build if you do not have the dependencies installed.

The C++ dependencies are checked at CMake time, right? We should just exclude the tests for which the deps are missing, otherwise we will get a lot of spurious failures.

XzzX commented:
> > > yeah, as long as the test don't fail, but get skipped everything is fine.
> > 
> > 
> > They will fail. But I think this is ok since you do not meet the dependencies. C++ will also not build if you do not have the dependencies installed.
> 
> The C++ dependencies are checked at CMake time, right? We should just exclude the tests for which the deps are missing, otherwise we will get a lot of spurious failures.

Why? That's how Python works. Everything is checked at runtime. Something like `could not find h5py` is a pretty clear indication that you do not meet all dependencies. Deactivating tests just gives you an incorrect feeling that everything is ok.

XzzX commented:
> I am thinking more from the C++ side, where `configure`, `make`, `make test` is the normal workflow. 

Yes, correct, and this works without a problem for the C++ side of the code, namely `_espressopp`.

Problems rise due to the hand-crafted integration of the Python frontend `espressopp` into cmake. Requiring hacks like `TEST_ENV` and calling the Python interpreter explicitly. Since `espressopp` is using `unittest` the correct way to integrate tests would be via `unittest discovery`, however, this does not work since `espressopp` does not follow the `unittest` guidelines.

Furthermore, a Python user should be used to missing package errors and know how to deal with them as this is classic Python business.

> I even have seen a couple of `setup.py` that check for dependencies and warn about reduced functionality. 

Yes, pythonic and very similar to my `requirements.txt`. Could be done but seems like no one cared about that.

> I think for our user community we have to make things as easy as possible hence I vote for a nice readable message at configure time.

That's not how Python works. As a Python user I do not know what `configure` is. I vote for implementing things in the ideology the programming language has. In my opinion this makes it easier for new users as the programming language behaves like they expect it. For Python this means, if you have an error in your script you will be notified at runtime, true for all Python errors.

junghans commented:
Go ahead, I am not the one that has to provide user support ;-)

XzzX commented on README.md:
```diff
@@ -29,14 +29,15 @@ $ git clone https://github.com/espressopp/espressopp.git
- - HDF5 library (>= 1.8) (optional for [H5MD](https://nongnu.org/h5md/) file format)
- - h5py library (>= 2.6.0) (optional for [H5MD](https://nongnu.org/h5md/) file format)
- - GROMACS library (required when `WITH_XTC` flag is enabled)
+#Dependencies
```
fixed
